pkg/services/ngalert/notifier/channels/discord.go

```diff
@@ -116,18 +134,33 @@ func (d DiscordNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool,
 		"icon_url": "https://grafana.com/assets/img/fav32.png",
 	}
 
-	embed := simplejson.New()
-	embed.Set("title", tmpl(DefaultMessageTitleEmbed))
-	embed.Set("footer", footer)
-	embed.Set("type", "rich")
+	linkEmbed := simplejson.New()
+	linkEmbed.Set("title", tmpl(DefaultMessageTitleEmbed))
+	linkEmbed.Set("footer", footer)
+	linkEmbed.Set("type", "rich")
 
 	color, _ := strconv.ParseInt(strings.TrimLeft(getAlertStatusColor(alerts.Status()), "#"), 16, 0)
-	embed.Set("color", color)
+	linkEmbed.Set("color", color)
 
 	ruleURL := joinUrlPath(d.tmpl.ExternalURL.String(), "/alerting/list", d.log)
-	embed.Set("url", ruleURL)
+	linkEmbed.Set("url", ruleURL)
 
-	bodyJSON.Set("embeds", []interface{}{embed})
+	embeds := []interface{}{linkEmbed}
+
+	attachments := d.constructAttachments(ctx, as, DiscordMaxEmbeds-1)
```

Comment:
We add a link to the alert list view as the first embed. So, one slot out of the max number of embeds is taken => `DiscordMaxEmbeds - 1`